fault/wrapper.py

```diff
@@ -19,29 +19,42 @@ def __setattr__(self, attr, value):
         # Hack to stage this after __init__ has been run, should redefine this
         # method in a metaclass? Could also use a try/except pattern, so the
         # exceptions only occur during object instantiation
+        try:
+            init_done = object.__getattribute__(self, "init_done")
+        except AttributeError:
+            init_done = False
+
+        if not init_done:
+            object.__setattr__(self, attr, value)
+            return
+
         if hasattr(self, "circuit") and hasattr(self, "instance_map"):
             if attr in self.circuit.interface.ports.keys():
                 if isinstance(self.parent, fault.TesterBase):
                     self.parent.poke(self.circuit.interface.ports[attr], value)
                 else:
                     raise NotImplementedError()
             else:
-                object.__setattr__(self, attr, value)
+                raise AttributeError(f"{attr} is not a valid port")
         else:
-            object.__setattr__(self, attr, value)
+            raise AttributeError(f"{attr} is not a valid port")
 
     def __getattr__(self, attr):
         # Hack to stage this after __init__ has been run, should redefine this
         # method in a metaclass?
         try:
-            object.__getattribute__(self, "init_done")
+            init_done = object.__getattribute__(self, "init_done")
+        except AttributeError:
+            init_done = False
+
+        if init_done:
             if attr in self.circuit.interface.ports.keys():
                 return PortWrapper(self.circuit.interface.ports[attr], self)
             elif attr in self.instance_map:
                 return InstanceWrapper(self.instance_map[attr], self)
             else:
                 object.__getattribute__(self, attr)
```

Comment:
Just noticing this is probably an error right?  I expect it should be `return object.__getattribute__(self, attr)`.

Reply:
yes it should be